app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -390,7 +390,8 @@ class HomeFragment : Fragment() {
 
         FxNimbus.features.homescreen.recordExposure()
 
-        displayWallpaperIfEnabled()
+        setWallpaperToCurrent()
```

Comment:
This read like it's a hack for something - we're reading from the store first and setting the wallpaper, before we start to observe the store and set the wallpaper again? What is happening that makes it slow to just have the observer which gives you the initial state?

Reply:
Ah, you're right. The repetition there shouldn't actually be necessary. Thanks!

Reply:
Actually, it looks like it is required to get the response time we'd like. I could only make guesses as to why the observation seems to take longer. I would guess it might relate to the flow being collected in a coroutine that is `launch`ed, which is why I had to switch to this `runBlocking` implementation in the first place.

I believe we'll still want the observation though, as we will want the home screen to react to changes that are made from the upcoming wallpaper selection tool https://github.com/mozilla-mobile/fenix/issues/26216. I don't have a good way to avoid these repeated calls of the top of my head, unless we want to start the observation from wherever we launch the selection tool from instead of from `onCreate`

Reply:
Come to think of it, this may just be related to `viewLifecycleOwner.lifecycleScope`. IIRC it doesn't run coroutines until after `onViewCreated`

Reply:
Changing it to the fragment's direct `lifecycleScope` or even the activity's `lifecycleScope` did not seem to improve the issue. 

I would be okay with removing the observation here for now and punting on finding the right place for it until we do the selection tool. The wallpaper should only ever be updated while the fragment is destroyed, since it can only be currently changed from the wallpaper settings. Previously, we had the additional constraint of it being changed when the logo on the home screen was clicked but we have removed that feature

Reply:
One way we can proceed here is to comment out the `observeWallpaperChanges` code for now and file a follow investigation. Perhaps @Mugurell or @mcarare will have a cycle to dig deeper and suggest a solution. Other than resolving the discussion here, I think the changes looks good from my perspective.

Reply:
I think that's a good idea. I will file a follow up

Reply:
Cool, I will pre-attemptively approve this for now since I don't need to re-review commented out changes. My only ask is that you throw it the newly filed issue into the current sprint.